bot.py

```diff
@@ -87,116 +91,166 @@ def translate_slack_usernames(self, unchecked_authors):
-        self.say("Wait, wait. Time out. My evil plan isn't evil enough until all the checkboxes are checked...")
-        await wait_for_checkboxes(self.org, self.repo, self.version)
+        await self.say("Wait, wait. Time out. My evil plan isn't evil enough until all the checkboxes are checked...")
+        await wait_for_checkboxes(self.org, self.repo, version)
         release_manager = release_manager_name()
-        self.say("All checkboxes checked off. Release {version} is ready for the Merginator{name}!".format(
-            version=self.version,
+        await self.say("All checkboxes checked off. Release {version} is ready for the Merginator{name}!".format(
+            version=version,
             name=' {}'.format(self.translate_slack_usernames([release_manager])[0]) if release_manager else '',
         ))
 
-    async def finish_release(self):
+    async def finish_release(self, version):
         """
         Merge the release candidate into the release branch, tag it, merge to master, and wait for deployment
         """
-        check_call([in_script_dir("finish_release.sh"), self.repo_dir, self.version])
-        self.say("Merged evil scheme {}! Now deploying to production...".format(self.version))
+        check_call([in_script_dir("finish_release.sh"), self.repo_dir, version])
+        await self.say("Merged evil scheme {}! Now deploying to production...".format(version))
         check_call([in_script_dir("wait_for_deploy.sh"), self.repo_dir, self.prod_hash_url, "release"])
-        self.say(
+        await self.say(
             "My evil scheme {} has been released to production. "
-            "And by 'released', I mean completely...um...leased.".format(self.version)
+            "And by 'released', I mean completely...um...leased.".format(version)
         )
 
-    def message_if_unchecked(self):
+    async def message_if_unchecked(self, version):
         """
         Send a message next morning if any boxes are not yet checked off
         """
-        unchecked_authors = get_unchecked_authors(self.org, self.repo, self.version)
+        unchecked_authors = get_unchecked_authors(self.org, self.repo, version)
         if unchecked_authors:
             slack_usernames = self.translate_slack_usernames(unchecked_authors)
-            self.say(
+            await self.say(
                 "What an unexpected surprise! "
                 "The following authors have not yet checked off their boxes: {}".format(
                     ", ".join(slack_usernames)
                 )
             )
 
+    async def delay_message(self, version):
+        """sleep until 10am next day, then message"""
+        now = datetime.now()
+        tomorrow_at_10 = next_workday_at_10(now)
+        await asyncio.sleep((tomorrow_at_10 - now).total_seconds())
+        await self.message_if_unchecked(version)
+
+    async def handle_message(self, content, loop):
+        """handle the message"""
+        if has_command(['release'], content):
```

Comment:
We should move the checks for `has_command(['start', 'release'], content)` and `has_command(['finish', 'release'], content)` up here and remove this check